### PR TITLE
Feature/load event

### PR DESCRIPTION
**Pull request by jackhkmatthews**
Implemented `imageLoaded` event chain

### CODE REVIEW

File: src/app/image-loader/image-loader.component.html [hunk 1]
```diff
@@ -6,7 +6,8 @@
   [class]="imgClass"
   inViewport
   (inViewportChange)="onInViewportChange($event)"
-  [debounce]="0">
+  [debounce]="0"
+  (load)="onImageLoad(image)">
```
Comment (edoparearyee) [inline src/app/image-loader/image-loader.component.html]: I think the argument for the event handler needs to be `$event` like on line 8

File: src/app/image-loader/image-loader.component.ts [hunk 2]
```diff
@@ -255,6 +276,14 @@ export class ImageLoaderComponent implements OnInit, AfterViewInit, OnDestroy {
     this.preloadSrc = '';
     this.loaded = true;
   }
+  /**
+   * When the main `img` element has loaded
+   *
+   * @memberof ImageLoaderComponent
+   */
+  public onImageLoad(image: ResponsiveImage): void {
+    this.imageLoaded.emit(image);
```
Comment (edoparearyee) [inline src/app/image-loader/image-loader.component.ts]: I was thinking do we need to emit the image load event when the placeholder is loaded? is it useful?

File: src/app/app.component.ts [hunk 3]
```diff
@@ -29,4 +29,22 @@ export class AppComponent {
       '@2x': 'http://via.placeholder.com/1400x800?text=lg+2x'
     }
   };
+
+  /**
+   * Incremented on each image load event.
+   *
+   * @type {number}
+   * @memberof ImageLoaderComponent
```
[the remark this answers is not in the document]
Comment (jackhkmatthews) [reply src/app/app.component.ts]: Corrected

File: src/app/app.component.html [hunk 4]
```diff
@@ -1,8 +1,13 @@
...
+<h3>Placeholder image loaded: <span class="placeholder-boolean">{{ placeholderLoaded }}</span></h3>
 <div class="spacer"></div>
+<h3>fullResLoaded event count: <span class="full-res-count">{{ fullResLoadedEventCount }}</span></h3>
 <sn-image-loader
   [image]="image"
   [sizes]="sizes"
   imgClass="foo"
+  alt="lorem ipsum"
+  (placeholderLoaded)="onPlaceholderLoad($event)"
+  (fullResLoaded)="onFullResLoad($event)"
```
Comment (edoparearyee) [inline src/app/app.component.html]: I would be very clear and specific and call these events `imageLoaded` and `imagePlaceholderLoaded` as `fullResLoaded` sounds ambiguous

File: src/app/image-loader/image-loader.component.ts [hunk 5]
```diff
@@ -134,6 +148,22 @@ export class ImageLoaderComponent implements OnInit, AfterViewInit, OnDestroy {
...
+  @Output()
+  public placeholderLoaded: EventEmitter<ImageLoadedEvent> = new EventEmitter<ImageLoadedEvent>();
+  /**
+   * Output for full res image loaded event.
+   *
+   * @type {EventEmitter}
+   * @memberof ImageLoaderComponent
+   */
+  @Output()
+  public fullResLoaded: EventEmitter<ImageLoadedEvent> = new EventEmitter<ImageLoadedEvent>();
```
Comment (edoparearyee) [inline src/app/image-loader/image-loader.component.ts]: Same here i would be clear and specific and call this `imageLoaded`

File: src/app/app.component.html (hunk 4 above)
Comment (jackhkmatthews) [reply src/app/app.component.html]: Changed

File: src/app/image-loader/image-loader.component.ts (hunk 5 above)
Comment (jackhkmatthews) [reply src/app/image-loader/image-loader.component.ts]: Changed

**Pull request merged**